plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/listeners/CodeWhispererCodeScanEditorMouseMotionListener.kt

```diff
@@ -391,4 +416,34 @@ class CodeWhispererCodeScanEditorMouseMotionListener(private val project: Projec
             )
         }
     }
+
+    private fun explainWithQ(issue: CodeWhispererCodeScanIssue) {
+        try {
+            // publish a message
+            project.messageBus.syncPublisher(
+                CodeScanActionsListener.SEND_CODESCAN_ISSUE_TO_Q
+            ).sendIssueToQ(issue.description.markdown, issue.codeText)
+            hidePopup()
+        } catch (err: Error) {
+            notifyError("Explain With Q failed")
+            LOG.error { "Q chat message publishing failed: $err" }
+        }
+    }
+
+    private fun fixWithQ(issue: CodeWhispererCodeScanIssue) {
+        try {
+            // publish a message
+            project.messageBus.syncPublisher(
+                CodeScanActionsListener.SEND_CODESCAN_ISSUE_TO_Q
+            ).fixIssueWithQ(issue.description.markdown, issue.codeText)
+            hidePopup()
+        } catch (err: Error) {
+            notifyError("Fix With Q failed")
+            LOG.error { "Q chat message publishing failed: $err" }
+        }
+    }
```

Comment:
if the Q window has not been previously opened, i dont believe this does anything

Reply:
I have for fix for this. I added startup activity for amazonq which pings chatController after startup and added window opening action to the functions.

Reply:
we should not be forcing the q window to open every time the user opens the IDE